src/blockdata/locktime/relative.rs

```diff
@@ -0,0 +1,281 @@
+    /// assert!(lock.is_satisfied_by(Time::default()));
+    ///
+    /// // Any time value satisfies a height-based lock.
+    /// let intervals: u16 = 70; // approx 10 hours;
+    /// let lock = Sequence::from_512_second_intervals(intervals).to_relative_lock_time().expect("valid time");
+    /// assert!(lock.is_satisfied_by(Height::default()));
+    /// ```
+    pub fn is_satisfied_by<T: SatisfiesLockTime>(self, satisfied_by: T) -> bool {
+        satisfied_by.satisfies_lock_time(self)
+    }
+}
+
+impl From<Height> for LockTime {
+    fn from(h: Height) -> Self {
+        LockTime::Blocks(h)
+    }
+}
+
+impl From<Time> for LockTime {
+    fn from(t: Time) -> Self {
+        LockTime::Time(t)
+    }
+}
+
+/// A relative lock time, block height value.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Height(u16);
+
+impl Height {
+    /// Returns the inner `u16` value.
+    pub fn value(&self) -> u16 {
+        self.0
+    }
+}
+
+impl From<u16> for Height {
+    fn from(value: u16) -> Self {
+        Height(value)
+    }
+}
+
+/// A relative lock time, time interval value.
+///
+/// For BIP 68 relative time locks, lock by time, time is measure in 512 second intervals.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Time(u16);
+
+impl Time {
+    /// Create a [`Time`] using time intervals where each interval is equivalent to 512 seconds.
+    ///
+    /// Encoding finer granularity of time for relative lock-times is not supported in Bitcoin.
+    #[inline]
+    pub fn from_512_second_intervals(intervals: u16) -> Self {
+        Time(intervals)
+    }
+
+    /// Create a [`Time`] from seconds, converting the seconds into 512 second interval with floor
+    /// division.
+    ///
+    /// # Errors
+    ///
+    /// Will return an error if the input cannot be encoded in 16 bits.
+    #[inline]
+    pub fn from_seconds_floor(seconds: u32) -> Result<Self, Error> {
+        if let Ok(interval) = u16::try_from(seconds / 512) {
+            Ok(Time::from_512_second_intervals(interval))
+        } else {
+            Err(Error::IntegerOverflow(seconds))
+        }
+    }
+
+    /// Create a [`Time`] from seconds, converting the seconds into 512 second interval with ceiling
+    /// division.
+    ///
+    /// # Errors
+    ///
+    /// Will return an error if the input cannot be encoded in 16 bits.
+    #[inline]
+    pub fn from_seconds_ceil(seconds: u32) -> Result<Self, Error> {
+        if let Ok(interval) = u16::try_from((seconds + 511) / 512) {
+            Ok(Time::from_512_second_intervals(interval))
+        } else {
+            Err(Error::IntegerOverflow(seconds))
+        }
```

Comment:
In fb5abb43711a9f39e62f0e7b25dda8f85b32e98b:

nit: this whole functtion could just call `from_seconds_floor(seconds + 511)`.

Reply:
Realized while implementing this that doing so would put the modified `seconds` into the error. I left as is for now. If we want to change we should do it here and in `transaction.rs` as well.

Reply:
Ok, that's fair.